src/extensions/php_fpm_metrics.go

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package extensions
+
+import (
+	"context"
+
+	agent_config "github.com/nginx/agent/sdk/v2/agent/config"
+	"github.com/nginx/agent/v2/src/core"
+	"github.com/nginx/agent/v2/src/core/config"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// Version
+	phpFpmMetricsExtensionPluginVersion = "v0.1.0"
+)
+
+type PhpFpmMetrics struct {
+	ctx       context.Context
+	ctxCancel context.CancelFunc
+	pipeline  core.MessagePipeInterface
+	env       core.Environment
+}
+
+func NewPhpFpmMetrics(env core.Environment, conf *config.Config) (*PhpFpmMetrics, error) {
+	return &PhpFpmMetrics{
+		env: env,
+	}, nil
+}
+
+func (m *PhpFpmMetrics) Init(pipeline core.MessagePipeInterface) {
+	log.Infof("%s initializing", agent_config.PhpFpmMetricsExtensionPlugin)
```

Comment:
future: consider storing a logger inside `PhpFpmMetrics`, and using [log fields](https://pkg.go.dev/github.com/sirupsen/logrus#readme-fields) to ensure every log message from our extension is identifiable and easy to filter on.

```go
logger := log.WithFields(log.Fields{"extension": agent_config.PhpFpmMetricsExtensionPlugin})
return &PhpFpmMetrics{
		env: env,
        logger: logger
	}
```

Similar to https://github.com/nginx/agent/blob/181b995a87a7c2905915f20acd0d2053c553ce64/src/extensions/nginx-app-protect/monitoring/manager/manager.go#L94

Reply:
Thanks Ryan. I will consider it in my following MRs on php module when I add more functionality